src/keyring.ts

```diff
@@ -169,6 +169,22 @@ export type Keyring<State extends Json> = {
     options?: Record<string, unknown>,
   ): Promise<string>;
 
+  /**
+   * Sign an EIP-7702 authorization. This is a signing method for authorizing a
+   * specific contract on a specific chain.
+   *
+   * @param address - The address of the account to use for signing.
+   * @param authorization - An array containing the chain ID, contract address,
+   * and nonce.
+   * @param options - Signing options; differs between keyrings.
+   * @returns The signed authorization as a hex string.
+   */
+  signEip7702Authorization?(
+    address: Hex,
+    authorization: [chainId: number, contractAddress: Hex, nonce: number],
```

Comment:
We could have used `@metamask/eth-sig-util` here too, but that would pull a new dep and since we plan to move this type to `accounts/keyring-utils`, I think that's fine that we keep it as is.